torchgeo/models/changestar.py

```diff
@@ -0,0 +1,229 @@
+
+    def forward(self, x: Tensor) -> Dict[str, Any]:
+        """Forward pass of the model.
+
+        Args:
+            x: a bitemporal input tensor of shape [B, T, C, H, W]
+
+        Returns:
+            a directory containing
+                if training stage, returning
+                    bi_seg_logit: bitemporal semantic segmentation logit
+                    bi_change_logit: bidirected binary change detection logit
+                if inference stage, returning
+                    bi_seg_logit: bitemporal semantic segmentation logit
+                    change_prob:  binary change detection probability
+        """
+        b, t, c, h, w = x.shape
+        x = rearrange(x, "b t c h w -> (b t) c h w")
+        # feature extraction
+        bi_feature = self.dense_feature_extractor(x)
+        # semantic segmentation
+        bi_seg_logit = self.seg_classifier(bi_feature)
+        bi_seg_logit = rearrange(bi_seg_logit, "(b t) c h w -> b t c h w", t=t)
+
+        bi_feature = rearrange(bi_feature, "(b t) c h w -> b t c h w", t=t)
+        # change detection
+        c12, c21 = self.changemixin(bi_feature)
+
+        results: Dict[str, Any] = {}
+        if not self.training:
+            results.update({"bi_seg_logit": bi_seg_logit})
+            if self.inference_mode == "t1t2":
+                results.update({"change_prob": c12.sigmoid()})
+            elif self.inference_mode == "t2t1":
+                results.update({"change_prob": c21.sigmoid()})
+            elif self.inference_mode == "mean":
+                results.update(
+                    {
+                        "change_prob": torch.stack([c12, c21], dim=0)
+                        .sigmoid_()
+                        .mean(dim=0)
+                    }
+                )
+            return results
+
+        results.update({"bi_seg_logit": bi_seg_logit, "bi_change_logit": [c12, c21]})
+        return results
+
+
+class ChangeStarFarSeg(ChangeStar):
+    """The network architecture of ChangeStar(FarSeg).
+
+    ChangeStar(FarSeg) is composited of a FarSeg model and a ChangeMixin module.
+
+    If you use this model in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/2108.07002
+    """
+
+    def __init__(
+        self,
+        backbone: str = "resnet50",
+        classes: int = 1,
+        backbone_pretrained: bool = True,
+    ) -> None:
+        """Initializes a new ChangeStarFarSeg model.
+
+        Args:
+            backbone: name of ResNet backbone
+                (default='resnet50')
+            classes: number of output segmentation classes
+                (default=1 for binary segmentation)
+            backbone_pretrained: whether to use pretrained weight for backbone
+                (default=True)
+        """
+        model = FarSeg(
+            backbone=backbone, classes=classes, backbone_pretrained=backbone_pretrained
+        )
+        seg_classifier: Any = model.decoder.classifier
```

Comment:
Wny `Any` and not `Module`?

Reply:
just for avoiding mypy issues.
That's Ok if `Module` can work.